src/modules/idyntree-yarp-visualizer/RobotConnectors.cpp

```diff
@@ -0,0 +1,558 @@
+
+        yarp::os::Bottle remoteControlBoards;
+        yarp::os::Bottle & remoteControlBoardsList = remoteControlBoards.addList();
+        for (auto& cb : m_controlBoards)
+        {
+            remoteControlBoardsList.addString("/" + m_basicInfo->robotPrefix + "/" + cb);
+        }
+        remapperOptions.put("remoteControlBoards",remoteControlBoards.get(0));
+        remapperOptions.put("localPortPrefix", "/" + m_basicInfo->name +"/remoteControlBoard:i");
+
+        if(!m_robotDevice.open(remapperOptions))
+        {
+            yError() << "Failed to open remote control board remapper.";
+            return false;
+        }
+        if (!m_robotDevice.view(m_encodersInterface) || !m_encodersInterface)
+        {
+            yError() << "Failed to view encoder interface.";
+
+            return false;
+        }
+    }
+
+    m_connected = true;
+    return true;
+}
+
+bool RemapperConnector::getJointValues(iDynTree::VectorDynSize &jointValuesInRad)
+{
+    if (m_connected)
+    {
+        std::lock_guard<std::mutex> lock(m_mutex);
+
+        if (m_encodersInterface->getEncoders(m_jointsInDeg.data()))
+        {
+            for (size_t i = 0; i < m_jointsInDeg.size(); ++i)
+            {
+                jointValuesInRad(i) = iDynTree::deg2rad(m_jointsInDeg(i));
+            }
+        }
+    }
+    return true;
+}
+
+void RemapperConnector::close()
+{
+    m_connected = false;
+    {
+        std::lock_guard<std::mutex> lock(m_mutex);
+        m_robotDevice.close();
+        m_encodersInterface = nullptr;
+    }
+}
+
+bool StateExtConnector::getAxesDescription(const yarp::os::Value &inputValue)
+{
+    m_cb_jointsMap.clear();
+    std::vector<std::string> jointNamesVector;
+
+    if (!inputValue.isList())
+    {
+        yError() << "connectToStateExt is specified, but the input is not a list.";
+        return false;
+    }
+
+    yarp::os::Bottle* cbJointList = inputValue.asList();
+
+    if (cbJointList->size() %2 != 0)
+    {
+        yError() << "The list provided after connectToStateExt is supposed to have an even number of elements."
+                 << "It needs to be a sequence of a list containing the name of the control board (and eventuually the number of joints to consider),"
+                     "followed by the full list of joints in the control board.";
+        return false;
+    }
+
+    for (size_t i = 0; i < cbJointList->size(); i += 2)
+    {
+        m_cb_jointsMap.emplace_back();
+        m_cb_jointsMap.back().jointsToConsider = -1;
+        std::vector<JointInfo>& jointVector = m_cb_jointsMap.back().joints;
+
+        yarp::os::Value& cbName = cbJointList->get(i);
+        if (cbName.isString())
+        {
+            m_cb_jointsMap.back().name = cbName.asString();
+
+        }
+        else if (cbName.isList())
+        {
+            yarp::os::Bottle* cbNameList = cbName.asList();
+            if (cbNameList->size() != 2 && cbNameList->size() != 1)
```

Comment:
```suggestion
            if (cbNameList->size() != 2 || cbNameList->size() != 1)
```

Reply:
Nope, it needs to be an &&. The error is printed if it is neither of size 1, nor 2.

Reply:
Right, my bad.